transaction/builder/src/memo_builder/rth_memo_builder.rs

```diff
@@ -532,6 +536,46 @@ mod tests {
         output_memo: Result<MemoPayload, NewMemoError>,
         change_memo: Result<MemoPayload, NewMemoError>,
     }
+    /// A function that returns a flexible memopayload for outputs
+    pub type FlexibleOutputMemoGenerator = Box<
+        dyn Fn(FlexibleMemoOutputContext) -> Result<FlexibleMemoPayload, NewMemoError>
+            + Sync
+            + Send,
+    >;
+    /// A function that returns a flexible memopayload for change
+    pub type FlexibleChangeMemoGenerator = Box<
+        dyn Fn(FlexibleMemoChangeContext) -> Result<FlexibleMemoPayload, NewMemoError>
+            + Sync
+            + Send,
+    >;
+
+    // Test memo generator that allows the use of closures
+    struct FlexibleMemoGeneratorClosure {
+        output_memo_generator: FlexibleOutputMemoGenerator,
+        change_memo_generator: FlexibleChangeMemoGenerator,
+    }
+
+    impl Debug for FlexibleMemoGeneratorClosure {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            f.debug_struct("FlexibleMemoGeneratorClosure").finish()
+        }
+    }
+
+    impl FlexibleMemoGeneratorTrait for FlexibleMemoGeneratorClosure {
+        fn create_output_memo(
+            &self,
+            context: FlexibleMemoOutputContext,
+        ) -> Result<FlexibleMemoPayload, NewMemoError> {
+            (self.output_memo_generator)(context)
+        }
+
+        fn create_change_memo(
+            &self,
+            context: FlexibleMemoChangeContext,
+        ) -> Result<FlexibleMemoPayload, NewMemoError> {
+            (self.change_memo_generator)(context)
+        }
+    }
```

Comment:
Remove the use of closures and just implement the methods directly on a new struct called something like `TestFlexibleMemoGenerator`. We shouldn't be using `Fn` trait or closures in this new pattern.

I also don't think we need to test the usage of closures since this is an implementation detail and tests shouldn't test implementation details. We can trust that the rust language's guarantees about composing methods from closures should just work.

Reply:
I actually wanted to keep this because I think it's helpful to have an example of creating a struct that implements the trait using closures. I agree that it's a little inelegant, but I think it should be equivalent for the purpose of the test. I can get rid of it if you feel strongly about it, but I do think it isn't introducing extra complexity entirely for no reason.

Reply:
I think I agree with Sam, this code is simpler and cleaner if you just implement functions on a struct as normal. 

> I can get rid of it if you feel strongly about it, but I do think it isn't introducing extra complexity entirely for no reason.

What is the reason? 

> helpful to have an example of creating a struct that implements the trait using closures

This is not a good reason to code things in a convoluted way. Creating examples like this is not directly related to any of mobilecoin's business or engineering goals afaict.